### PR TITLE
Gardener3.0

### PR DESCRIPTION
**Pull request by exxaxxxx**
Improved gardener, spawns in direction to enemy, fixed number of scouts spawned. Adjusted Archon and Robot Victory points buying procedure to allow faster expand and yet be able to win with victory points. Changed lot of constants to build more gardeners and overall improve economics. Damaged trees are watered first.

I cleared all stack trace from console, it became useless for me and it slowed down the whole game. Can be uncommented to get them back. Fixed some edge cases (i.e. when Soldier was trying to move in null direction).

TODO no interaction with soldiers.

I apologize for some code formatting changes, Idea did it 😔 reject these changes if needed.

Overall this branch is in state where everything somehow works and the AI is both economically and offensively OK.

### CODE REVIEW

File: src/blueteam/Archon.java [hunk 1]
```diff
@@ -12,27 +12,56 @@
...
 		}
-
-		// randomly attempt to build a gardener if we need more
+		// Hardcoded first 180 rounds. Just build gardener each 60 rounds.
+		// Easiest way to give gardeners enough time to expand. After round 180 continue in general approach.
 		Direction dir = randomDirection();
-		System.out.println(getRobotCount(RobotType.GARDENER));
-		if (rc.canHireGardener(dir) && TeamConstants.DESIRED_NUMBER_OF_GARDENERS > getRobotCount(RobotType.GARDENER)) {
+		if (rc.getRoundNum() >= 0 && rc.getRoundNum() < 60 && step == 0) {
+			rc.hireGardener(dir);
```
Comment (Formicarufa) [inline src/blueteam/Archon.java]: It surprises me that we do not check if the hireGardener was successful (canHireGardener was removed) and that the behaviour does not change according to the number of archons we have. But if it works, I have no objections.

**Comment by Formicarufa** [comment]
In certain scenarios, the gardeners completely fail and do not build garden despite having enough space, e.g. HouseDivided, Interference, Caterpillar, Conga.

**Comment by exxaxxxx** [comment]
@Formicarufa Fixed this gardener bug, improved gameplay and edge cases (Cramped).

**Review by hrnr — APPROVED** [review]
Works really nicely for me, much improved version. Good work. Some spaghetti there and there, but that's ok with me, project is nearing its end, and we already have some italian spirit in the master, so this will fit nicely.

**Pull request merged**